genie/database_to_staging.py

```diff
@@ -442,6 +447,64 @@ def runMAFinBED(syn, center_mappingdf, test=False, genieVersion="test"):
     #     " is False and Center in ('{}')".format(
     #         mutationSynId, "','".join(center_mappingdf.center)))
     # removedVariantsDf = removedVariants.asDataFrame()
+    removed_variantsdf = store_maf_in_bed_filtered_variants(
+        syn=syn,
+        notinbed_file=notinbed_file,
+        center_mapping_df=center_mappingdf,
+        genie_version=genieVersion,
+    )
+    return removed_variantsdf["removeVariants"]
+
+
+def get_run_maf_in_bed_script_cmd(
+    notinbed_file: str, script_dir: str, test: bool, staging: bool
+) -> str:
+    """This function gets the MAFinBED R script command call based on
+       whether we are running in test, staging or production mode
+
+    Args:
+        notinbed_file (str): Full path to the notinbed csv
+        script_dir (str): directory of the MAFinBED script
+        test (bool): Testing parameter.
+        staging (bool): Staging parameter
+
+    Returns:
+        str: Full command call for the MAFinBED script
+    """
+    mafinbed_script = os.path.join(script_dir, "../R/MAFinBED.R")
+    # The MAFinBED script filters out the centers that aren't being processed
+    command = ["Rscript", mafinbed_script, notinbed_file]
+    if test:
+        command.append("--testing")
+    elif staging:
+        command.append("--staging")
+    else:
+        pass
+    return command
+
+
+def store_maf_in_bed_filtered_variants(
+    syn: synapseclient.Synapse,
+    notinbed_file: str,
+    center_mapping_df: pd.DataFrame,
+    genie_version: str,
+) -> pd.DataFrame:
+    """This script retrieves and stores the filtered variants generated
+        from running the MAFinBed script as files on Synapse. This script
+        then returns the not in bed file with the removed variants column
+        added.
+        TODO: Add handling for empty file?
```

Comment:
Good catch!  Is the notinbed_file sometimes empty?

Reply:
I don't think so, that would mean we have 0 variants that don't overlap with any of the bed regions ... ? How likely that is genomically I wouldn't know but if the code is broken, something like that could happen

Reply:
Very unlikely but good to think about